### ISSUE TITLE
Is there anyway to restrict user from creating new cat when "Input type: Input Text" for cat field.

### ISSUE THREAD

**Issue by kabirbd89** [weDevsOfficial/wp-user-frontend#1343]
In my site there is 600+ category and 900+ user. Its painful for user to select the specific category they supposed to post. But for 600 cat its tough to find out the specific category they are are try to post. 

I enable "Input type: Input Text" for category input field. so that user can write the first one or two character of cat and select the category from suggestion based on there search. 

Now the pain part, 
If user put any text in category field and there is no existing category, a new category will be created. I assume this ability giver by default **is there anyway to restrict user from creating new cat when "Input type: Input Text"**

Is there anyway to stop user being creating category but chose from an existing one. 

TIA

**Comment by anik-fahmid** [comment]
@kabirbd89 yes the ability is given by default and you cannot restrict it. We will review the scope and try to add an option for the admin. For now, you can use the Select option for Type in the Category fields (I understand that you have lots of categories).

**Comment by kabirbd89** [comment]
@anik-fahmid I actually maintaining a non profit website for a community, I cant go to some professional for that reason. On the other hand I am complete outside of coding world.

From my basic experience I have an idea can you please give me a reply based on my idea. 

1. creating a custom field (category type) for user profile and assign specific category to specific user. With the help of ACF
2. And then use that custom field meta (unique_category) and then field value in the process of post submission. Its actually working when i add post from wp default dashboard. 

**But in wpuf form its not working. Can you please tell me the location of post submission hook i mean which file i need to edit.** 

Currently I add these code to theme function.php its work good. 

`function assign_category_to_post($post_id) {
    if (defined('DOING_AUTOSAVE') && DOING_AUTOSAVE) return;
    
    // Check if it's a new post being created
    if ($post_id && get_post_type($post_id) === 'post' && get_post_status($post_id) === 'publish') {
        $user_id = get_current_user_id();
        
        // Get the user's selected category from ACF
        $selected_category = get_field('unique_category', 'user_' . $user_id);
        
        if ($selected_category) {
            // Convert the selected category to an array
            $selected_category = array($selected_category);
            
            // Set the post's categories
            wp_set_post_categories($post_id, $selected_category);
        }
    }
}
add_action('save_post', 'assign_category_to_post');`

**Comment by kabirbd89** [comment]
@anik-fahmid  thank you for your response. I am waiting for this feature.
